File: subjects/2023-2024-RO_VARIANT_I/alg_prog/Java/src/encoder/CircularEncoding.java
package encoder;

public class CircularEncoding extends Encoder {
    final int DEFAULT_CIRCULAR_ENCODING_LENGTH = 3;
    private final int length;

    public CircularEncoding(int length) {
        super();
        this.length = length > 2 ? length : DEFAULT_CIRCULAR_ENCODING_LENGTH;
    }

    @Override
    public String encode(String m) {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < m.length(); i++) {
            result.append(m.charAt((i + this.length) % m.length()));
        }

        return result.toString();
    }
}

File: subjects/2023-2024-RO_VARIANT_I/alg_prog/CSharp/CSharp/encoder/CircularEncoding.cs
namespace CSharp.encoder;

public class CircularEncoding : Encoder
{
    private const int DEFAULT_CIRCULAR_ENCODING_LENGTH = 3;
    private readonly int length;

    public CircularEncoding()
    {
        this.length = DEFAULT_CIRCULAR_ENCODING_LENGTH;
    }

    public CircularEncoding(int length)
    {
        this.length = length > 2 ? length : DEFAULT_CIRCULAR_ENCODING_LENGTH;
    }

    public override string encode(string m)
    {
        string result = "";

        for (var i = 0; i < m.Length; i++)
        {
            result += m[(i + length) % m.Length];
        }

        return result;
    }
}